rpc/core/mempool.go

```diff
@@ -42,6 +46,10 @@ func BroadcastTxSync(ctx *rpctypes.Context, tx types.Tx) (*ctypes.ResultBroadcas
 	}
 	res := <-resCh
 	r := res.GetCheckTx()
+
+	env.EventBus.PublishEventPendingTx(types.EventDataTx{TxResult: types.TxResult{
+		Tx: tx,
+	}})
```

Comment:
It is best to judge whether the execution result is successful, and then send the event if it succeeds.

Reply:
the code has been moved into mempool.addTx()